src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerOptions.cs

```diff
@@ -37,5 +38,17 @@ public AnalyzerOptions WithAdditionalFiles(ImmutableArray<AdditionalText> additi
 
             return new AnalyzerOptions(additionalFiles);
         }
+
+        public override bool Equals(object obj)
+        {
+            var other = obj as AnalyzerOptions;
```

Comment:
can we check whether two AnalyzerOptions has same reference equality first and then see whether AdditionalFiles has same and then each items.